lib/bolt/transport/local.rb

```diff
@@ -65,27 +60,298 @@ def run_command(target, command, _options = {})
         end
       end
 
+      ## DUP START
+      PS_ARGS = %w[
+        -NoProfile -NonInteractive -NoLogo -ExecutionPolicy Bypass
+      ].freeze
+
+      def powershell_file?(path)
+        Pathname(path).extname.casecmp('.ps1').zero?
+      end
+
+      def process_from_extension(path)
+        case Pathname(path).extname.downcase
+        when '.rb'
+          [
+            'ruby.exe',
+            ['-S', "\"#{path}\""]
+          ]
+        when '.ps1'
+          [
+            'powershell.exe',
+            [*PS_ARGS, '-File', "\"#{path}\""]
+          ]
+        when '.pp'
+          [
+            'puppet.bat',
+            ['apply', "\"#{path}\""]
+          ]
+        else
+          # Run the script via cmd, letting Windows extension handling determine how
+          [
+            'cmd.exe',
+            ['/c', "\"#{path}\""]
+          ]
+        end
+      end
+
+      def escape_arguments(arguments)
+        arguments.map do |arg|
+          if arg =~ / /
+            "\"#{arg}\""
+          else
+            arg
+          end
+        end
+      end
+
+      def shell_init
+        command = <<-PS
```

Comment:
I'm guessing this is copied from WinRM. Can we pull shared knowledge around how to run things in powershell out into a separate class that the WinRM and Local transports can share?

Reply:
That would also be useful if we eventually support powershell over SSH. Like a transport vs. runtime separation

Reply:
Will absolutely refactor to use a shared class for powershell code. I was just trying to get something working to write tests against. Having tests will help me with the refactor.